src/backend/access/appendonly/aosegfiles.c

```diff
@@ -49,6 +49,13 @@
 #include "utils/numeric.h"
 #include "utils/visibility_summary.h"
 
+#define VALIDATE_GP_ROLE() \
+	if (Gp_role != GP_ROLE_DISPATCH) \
+		ereport(ERROR,\
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),\
+				 errmsg("function '%s' must be executed on the dispatcher",\
+						PG_FUNCNAME_MACRO)));
```

Comment:
Add "DO WHILE" to this macro to avoid problem if we write:
```
if (something)
  VALIDATE_GP_ROLE();
else
   do_things
```

Reply:
> Add "DO WHILE" to this macro to avoid problem if we write

Ok.

That could have been avoided if all single statement bodies of conditional/loop statements were wrapped in braces.
It is a mandatory requirement for many mission critical systems (for ex. refer to https://pvs-studio.com/en/docs/warnings/v2507/ ).
Unfortunately, it is not commonly used in postgres code :(.